devices/BlueNrg2/SpiInterface.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Threading;
+using BlueNrg2;
+using Microsoft.Extensions.Logging;
+
+namespace Iot.Device.BlueNrg2
+{
+	internal class SpiInterface : IHardwareInterface
+	{
+		private readonly ILogger _logger;
```

Comment:
most likely you can apply the same pattern than for the constructor and have the logger only when build in DEBUG.